io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter.java

```diff
@@ -1766,11 +1801,17 @@ private void writeElement(Object value, PropertyDefinition propDef, IOReporter r
 				boolean writeElements = !hasValue && !hasOnlyNilReason;
 				boolean isNil = !writeElements && (!hasValue || value == null);
 
-				// write all children
-				writeProperties(group, group.getDefinition(), writeElements, isNil, report);
+				/*
+				 * // write all children writeProperties(group,
```

Comment:
Changing the logic here looks to me like something that breaks a lot of things. Child properties should be handled even if there is no value present directly in the parent.

Reply:
@Flaminietta the test failures probably come from this, but there are also other issues. I'll prepare an update of the PR.